### DISCUSSION TITLE
Email processing Question

### DISCUSSION THREAD

**Discussion by gismcieri** [ecederstrand/exchangelib#936] [General]
When you do a filter such as:
mail2process = account.inbox.filter(datetime_received__range=(start, end),subject__contains='Miss',is_read=False )

Does exchagelib download all the results or does it do a new query for every record you loop through to get the information.  

For example if i get a return of a 100 emails and i loop through the emails to get the attachment is that on query query or a 101 queries.

Thanks,
Matt

**Comment by ecederstrand** [comment]
That depends on the paging values you set on the query. By default we get 100 items per page.

if you enable debug logging (see https://ecederstrand.github.io/exchangelib/#troubleshooting) then you can see exactly what’s going on.
